src/graphics/aurora/model_kotor.cpp

```diff
@@ -650,6 +744,51 @@ void ModelNode_KotOR::readMesh(Model_KotOR::ParserContext &ctx) {
 	createBound();
 
 	ctx.mdl->seek(endPos);
+
+	_mdxStructSize = mdxStructSize;
+	_vertexCount = vertexCount;
+	_offNodeData = offNodeData;
+}
+
+void ModelNode_KotOR::readSkin(Model_KotOR::ParserContext &ctx) {
+	ctx.mdl->skip(12);
+	uint32 mdxOffsetBoneWeights = ctx.mdl->readUint32LE();
+	uint32 mdxOffsetBoneMappingId = ctx.mdl->readUint32LE();
+	uint32 boneMappingOffset = ctx.mdl->readUint32LE();
+	uint32 boneMappingCount = ctx.mdl->readUint32LE();
+	ctx.mdl->skip(72);
+
+	_mesh->skin = new Skin();
+	_mesh->skin->boneMapping = new float[boneMappingCount];
+	_mesh->skin->boneMappingCount = boneMappingCount;
+	_mesh->skin->boneWeights = new float[_vertexCount * 4];
+	_mesh->skin->boneMappingId = new float[_vertexCount * 4];
+
+	uint32 pos = ctx.mdl->seek(ctx.offModelData + boneMappingOffset);
+	for (uint32 i = 0; i < boneMappingCount; i++)
+		_mesh->skin->boneMapping[i] = ctx.mdl->readIEEEFloatLE();
+	ctx.mdl->seek(pos);
```

Comment:
This should go back to the position after the skin header(where you are after line 759) . You cant assume that the start of the bone mapping matches the end of the skin header.

Reply:
Not sure I understand you there. It doesn't read bone data from the end of the skin header. First it moves to the offset specified in skin header, reads the data, then returns to previous position.

Reply:
The previous position you go back to is the start of the bone data.
The previous position you should go back is the end of the skin header.

Reply:
According to the seek method description and your own binary template I'm pretty sure it goes back to the end of skin header as is.

![2017-11-18_20-47-42](https://user-images.githubusercontent.com/7695218/32980926-ce783b08-cca1-11e7-9349-691d519359ce.png)
![2017-11-18_20-42-22](https://user-images.githubusercontent.com/7695218/32980925-cba3221c-cca1-11e7-968c-e3c338432c45.png)

Reply:
You're right. I didn't realize seek returns the previous position before the seek.